pkg/kcp/provider/gcp/vpcpeering/client/cloudComputeClient.go

```diff
@@ -45,6 +53,20 @@ func (c *networkClient) CreateVpcPeeringConnection(ctx context.Context, name *st
 	kymaNetwork := getFullNetworkUrl(*kymaProject, *kymaVpc)
 	remoteNetwork := getFullNetworkUrl(*remoteProject, *remoteVpc)
 
+	//NetworkPeering will only be created if the remote vpc has a tag with the kyma shoot name
+	remoteNetworkInfo, err := c.cnc.Get(ctx, &pb.GetNetworkRequest{Network: *remoteVpc, Project: *remoteProject})
+	if err != nil {
+		return nil, err
+	}
+
+	isRemoteNetworkTagged, err := c.CheckRemoteNetworkTags(ctx, remoteNetworkInfo, *kymaVpc)
+
+	if isRemoteNetworkTagged == false || (err != nil && err.Error() == "no more items in iterator") {
+		return nil, fmt.Errorf("remote network " + *remoteVpc + " is not tagged with the kyma shoot name " + *kymaVpc)
+	} else if err != nil {
+		return nil, err
+	}
+
```

Comment:
I know this is for your prior PR but I just saw it.
We should never wait for an async request to finish. Each loop should finish as soon as possible. This can cause many issues:
err = peeringOperationFromKyma.Wait(ctx)

Instead, return the operation, keep it in status of your object (on k8s), and when looping, check the operation and see when it is done and if it has any errors. All other CRs follow that model. You can check them out for an example.